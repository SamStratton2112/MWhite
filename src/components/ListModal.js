import List from './List';
import { useState } from 'react';
import Button from 'react-bootstrap/Button';
import Modal from 'react-bootstrap/Modal';

const ListModal = ({list, name})=>{
  const [show, setShow] = useState(false);

  const handleClose = () => setShow(false);
  const handleShow = () => setShow(true);

  return (
    <>
      <Button variant="primary" onClick={handleShow}>
        {name}
      </Button>

      <Modal
        show={show}
        onHide={handleClose}
        backdrop="static"
        keyboard={false}
      >
        <Modal.Header closeButton>
          <Modal.Title>{name}</Modal.Title>
        </Modal.Header>
        <Modal.Body>
          <List list={list}/>
        </Modal.Body>
        <Modal.Footer>
          <Button variant="secondary" onClick={handleClose}>
            Close
          </Button>
        </Modal.Footer>
      </Modal>
    </>
  );
};

export default ListModal;