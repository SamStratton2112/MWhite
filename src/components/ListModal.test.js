import ListModal from './ListModal';
import { render } from '@testing-library/react';

const testList = ['test item 1', 'test item 2'];

describe(ListModal, () => {
    it("renders without crashing", () => {
        render(
            <ListModal list={testList}/>);
    });
    it("matches snapshot", () => {
        const { asFragment } = render(
            <ListModal list={testList}/>);
        expect(asFragment()).toMatchSnapshot();
    });
});