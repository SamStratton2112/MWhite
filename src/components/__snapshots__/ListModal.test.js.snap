// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ListModal matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="bttn"
  />
</DocumentFragment>
`;
